vscode-dotnet-runtime-library/src/Utils/FileUtilities.ts

```diff
@@ -247,31 +251,32 @@ export class FileUtilities extends IFileUtilities
      *
      * @returns true if the process is running with admin privileges
      */
-    public isElevated(eventStream?: IEventStream): boolean
+    public async isElevated(context: IAcquisitionWorkerContext, utilContext: IUtilityContext): Promise<boolean>
     {
+        const executor = new CommandExecutor(context, utilContext);
         if (os.platform() !== 'win32')
         {
             try
             {
-                const commandResult = proc.spawnSync('id', ['-u']);
-                return commandResult.status === 0;
+                const commandResult = await executor.execute(CommandExecutor.makeCommand('id', ['-u']), { dotnetInstallToolCacheTtlMs: SYSTEM_INFORMATION_CACHE_DURATION_MS });
```

Comment:
This could be SYS_CMD_SEARCH_CACHE_DURATION_MS, but this is gonna be wiped when vscode restarts anyways which would be required to elevate it. I guess that's one bad caveat if someone upgrades the cache to be persistent in the future.